File: test/cpp/service/findnodebyid.cpp
#include <RobotRaconteur.h>

using namespace RobotRaconteur;
using namespace std;


int main(int argc, char* argv[])
{
    if (argc < 3)
    {
        cout << "Usage for findnodebyid:  findnodebyid id schemes" << endl;
        return -1;
    }

    RobotRaconteurNode::s()->SetLogLevelFromEnvVariable();

    NodeID id = NodeID(string(argv[1]));
    vector<string> schemes;
    boost::split(schemes, argv[2], boost::is_from_range(',', ','));

    RR_SHARED_PTR<TcpTransport> t = RR_MAKE_SHARED<TcpTransport>();
    t->EnableNodeDiscoveryListening();
    RobotRaconteurNode::s()->RegisterTransport(t);

    RR_SHARED_PTR<LocalTransport> c2 = RR_MAKE_SHARED<LocalTransport>();
    RobotRaconteurNode::s()->RegisterTransport(c2);

    RR_SHARED_PTR<HardwareTransport> c5 = RR_MAKE_SHARED<HardwareTransport>();
    RobotRaconteurNode::s()->RegisterTransport(c5);

    boost::this_thread::sleep(boost::posix_time::seconds(6));

    vector<NodeInfo2> r = RobotRaconteurNode::s()->FindNodeByID(id, schemes);

    for (vector<NodeInfo2>::iterator e = r.begin(); e != r.end(); e++)
    {
        cout << "NodeID: " << e->NodeID.ToString() << endl;
        cout << "NodeName: " << e->NodeName << endl;
        cout << "ConnectionURL: " << boost::join(e->ConnectionURL, ", ") << endl;
        cout << endl;
    }

    RobotRaconteurNode::s()->Shutdown();

    return 0;
}

File: test/python/RobotRaconteurTest/findnodebyid.py
from RobotRaconteur.Client import *
import sys
import time


def main():
    rrid = RR.NodeID(sys.argv[1])
    transports = sys.argv[2].split(",")

    time.sleep(6)

    ret = RRN.FindNodeByID(rrid, transports)

    for r in ret:
        print_NodeInfo2(r)

    def printret(ret):
        for r in ret:
            print_NodeInfo2(r)
    RRN.AsyncFindNodeByID(rrid, transports, printret, 1)

    time.sleep(2)


def print_NodeInfo2(s):
    print("NodeID: " + str(s.NodeID))
    print("NodeName: " + s.NodeName)
    print("ConnectionURL: " + str(s.ConnectionURL))


if __name__ == "__main__":
    main()
